Phaethon/Tests/IntegrationTests/LoginTest.cs

```diff
@@ -0,0 +1,202 @@
+
+            Cleanup();
+        }
+
+        [Test]
+        public async Task Login_NullPasswordLoginInfo_BadRequest()
+        {
+            //Setup
+            InitializeData();
+            
+            //Act
+            userModel2.Password = null;
+            var response = await _client.PostAsJsonAsync("Login/Login", userModel2);
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
+
+            Cleanup();
+        }
+
+        [Test]
+        public async Task Login_NullUserNameLoginInfo_BadRequest()
+        {
+            //Setup
+            InitializeData();
+
+            //Act
+            userModel2.Username = null;
+            var response = await _client.PostAsJsonAsync("Login/Login", userModel2);
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
+
+            Cleanup();
+        }
+        #endregion
+
+        #region Delete
+
+        [Test]
+        public async Task Delete_CorrectLoginInfo_OK()
+        {
+            // Setup
+           
+                InitializeData();
+           
+            //Act
+           
+            var response = await _client.PostAsJsonAsync("Login/Delete", userModel2.ID.ToString());
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+        }
+        #endregion
+
+        #region Add
+        [Test]
+        public async Task New_User_OK()
+        {
+            // Setup
+            InitializeData();
+
+            //Act
+            using (var db = new DatabaseContext())
+            {
+                count1 = db.Login.Count();
+                db.Login.Add(userModel);
+                db.SaveChanges();
+            }
+            using (var db = new DatabaseContext())
+            {
+                count2 = db.Login.Count();               
+            }
+            var response = await _client.PostAsJsonAsync("Login/CreateOrUpdate", userModel);
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.AreNotEqual(count1, count2);
+            loginDa = new LoginDa { };
+            using (var db = new DatabaseContext())
+            {
+                Login deleted = db.Login.Attach(userModel);
+                loginDa.Delete(db, deleted);
+            }
+            Cleanup();
+        }
+        #endregion
+
+        #region Edit
+        [Test]
+        public async Task Edit_User_OK()
+        {
+            // Setup
+            InitializeData();
+            name = userModel2.Username;
+            userModel2.Username = "Maybe";
+            name2 = userModel2.Username;
+            //Act
+            var response = await _client.PostAsJsonAsync("Login/CreateOrUpdate", userModel2);
+            //Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.AreNotEqual(name, name2);
+            Cleanup();
+        }
```

Comment:
You should add a test for what happens if someone tries to edit their username, but that username already exists in the database.

Normaly, the edit should fail and user should receive a BadRequest with friendly message on why that is not allowed.

Reply:
this comment still stands